### PR TITLE
312 create new event

### PR DESCRIPTION
**Pull request by zarrellab**
This PR closes #[312]

Separates an event's date and time out to date, start time, and end time.
-CreateEvent validates time input and shows error to user
-Separates visualization of times

[https://media.giphy.com/media/JIX9t2j0ZTN9S/giphy.gif]

### CODE REVIEW

**Comment by nlkluth** [comment]
noticing two issues. up to you if you want to follow up in a different PR.

1) Enter start time 12:00 AM. Enter end time 1:00 PM. Change both fields to include bad values (see screenshot). Form no longer works, including when I try to fix bad values.

2) Submitting the form with these bad values does not route, but _does_ create the event. Pushing "Schedule" multiple times ended up with multiple events.

image

**Comment by zarrellab** [comment]
CreateEvent action now merges the datetimes into original data format, instead of separating out dates and times. This should keep things more persistent with how they were already working.

**Comment by zarrellab** [comment]
As a note:
redux-form was updated to 6.8.0

**Comment by zarrellab** [comment]
I see that there are proptype errors for date objects. It claims that they are numbers. Should I change the proptype to number, instead of string?

**Review by nlkluth — APPROVED** [review]
Nice job! I didn't see date object/number propType warnings. Looks good to me otherwise

File: client/src/actions/eventActions.js [hunk 1]
```diff
@@ -6,8 +6,19 @@ import { correctDatesForKeys } from '../utils/dateFormatting';
 import { apiAction } from './modelActions';
 
 export function createEvent(formData) {
+  const postData = correctDatesForKeys(formData, ['date', 'startTime', 'endTime']);
+  postData.date.set({
```
Comment (nlkluth) [inline client/src/actions/eventActions.js:10]: I am not sure I follow this correctly. Looks like `date` is being passed into `correctDatesForKeys` and then immediately overridden. Maybe some room for refactoring here

**Pull request merged**